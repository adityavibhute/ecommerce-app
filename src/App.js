import { Routes, Route } from "react-router-dom"
import { useReducer } from "react";
import './App.css';
import Navigation from './components/Navigation';
import Footer from './components/Footer';
import { initialState, updateProducts } from "./components/Store/store.reducers";
import HeroBanner from './components/HeroBanner';
import ProductCategory from './components/ProductCategory';
import ProductSpecific from './components/ProductSpecific';
import CartList from './components/CartList';
import UserContext from "./UserContext";

const App = () => {
  const [productDetails, dispatch] = useReducer(updateProducts, initialState);

  return (
    <div data-testid="app-test-id" className="App">
      <UserContext.Provider value={{ dispatch, productDetails }}>
        <Navigation />
        <Routes>
          <Route path="/" element={<HeroBanner />} />
          <Route path="/products">
            <Route index element={<ProductCategory />} />
            <Route path=":id" element={<ProductSpecific />} />
          </Route>
          <Route path="/categories">
            <Route path="men's clothing">
              <Route index element={<ProductCategory />} />
              <Route path=":id" element={<ProductSpecific />} />
            </Route>
            <Route path="women's clothing">
              <Route index element={<ProductCategory />} />
              <Route path=":id" element={<ProductSpecific />} />
            </Route>
            <Route path="electronics">
              <Route index element={<ProductCategory />} />
              <Route path=":id" element={<ProductSpecific />} />
            </Route>
          </Route>
          <Route path="cart" element={<CartList />} />
        </Routes>
        <Footer />
      </UserContext.Provider>
    </div>
  );
}

export default App;
