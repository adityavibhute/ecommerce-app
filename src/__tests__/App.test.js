import {render, screen} from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom';

import App from '../App';
describe('APP components', () => {
    it('should test App component', () => {
        render(
            <Router>
              <App />
            </Router>
        );
        // Query for an element on the home page and assert its presence
        const homeElement = screen.getByText('Home');
        expect(homeElement).toBeInTheDocument();
    });

    it('renders products page correctly', () => {
        render(
          <Router initialEntries={['/products']}>
            <App />
          </Router>
        );
        const productRoute = screen.getByText('Products');
        expect(productRoute).toBeInTheDocument();
    });

    it('renders categories page correctly', () => {
        render(
          <Router initialEntries={['/categories']}>
            <App />
          </Router>
        );
        const categoryRoute = screen.getByText('Products');
        expect(categoryRoute).toBeInTheDocument();
    });
});