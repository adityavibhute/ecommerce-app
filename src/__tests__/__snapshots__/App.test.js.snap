// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP components should test App component 1`] = `
<div
  className="App"
>
  <ContextProvider
    value={
      Object {
        "dispatch": [Function],
        "productDetails": Object {
          "products": Array [],
          "total": 0,
        },
      }
    }
  >
    <Navigation />
    <Routes>
      <Route
        element={<HeroBanner />}
        path="/"
      />
      <Route
        path="/products"
      >
        <Route
          element={<ProductCategory />}
          index={true}
        />
        <Route
          element={<ProductSpecific />}
          path=":id"
        />
      </Route>
      <Route
        path="/categories"
      >
        <Route
          path="men's clothing"
        >
          <Route
            element={<ProductCategory />}
            index={true}
          />
          <Route
            element={<ProductSpecific />}
            path=":id"
          />
        </Route>
        <Route
          path="women's clothing"
        >
          <Route
            element={<ProductCategory />}
            index={true}
          />
          <Route
            element={<ProductSpecific />}
            path=":id"
          />
        </Route>
        <Route
          path="electronics"
        >
          <Route
            element={<ProductCategory />}
            index={true}
          />
          <Route
            element={<ProductSpecific />}
            path=":id"
          />
        </Route>
      </Route>
      <Route
        element={<CartList />}
        path="cart"
      />
    </Routes>
    <Footer />
  </ContextProvider>
</div>
`;
